asm2/main.js

```diff
@@ -0,0 +1,49 @@
+function openForm() {
+    document.getElementById("myForm").style.display = "block";
+  }
+  
+function closeForm(){
+    document.getElementById("myForm").style.display = "none";
+}
+
+function onFormSubmit() {
+    var formData = readFormData();
+    insertNewRecord(formData);    
+}
+function calTotal(){
+    var formData = readFormData();
+    calTotal(formData)
+}
+function readFormData() {
+    var formData = {};
+    formData["ProdName"] = document.getElementById("ProdName").value;
+    formData["price"] = document.getElementById("price").value;
+    formData["quantity"] = document.getElementById("quantity").value;
+    formData["date"] = document.getElementById("date").value;
+    return formData;
+}
+
+function insertNewRecord(data) {
+    // resetForm();
+    var table = document.getElementById("productList").getElementsByTagName('tbody')[0];
```

Comment:
Redundant call of method, choose 1 of 2
```suggestion
    let table = document.getElementById('productList');
    or
     let table = document.getElementsByTagName('tbody')[0];
```